### PR TITLE
Search data

### PR DESCRIPTION
**Pull request by rawalparas**
I created an API to search the book data using the name of any attribute of search collection. (name , author , language)

### CODE REVIEW

File: src/api/controller/create/createController.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ const authorModel = require("../../../model/authorModel.js");
 const bookModel = require("../../../model/bookModel.js");
 const languageModel = require("../../../model/languageModel.js");
 const searchModel = require('../../../model/searchModel.js');
+const mongoose = require('mongoose');
```
Comment (athanbhanit) [inline src/api/controller/create/createController.js:5]: Why have you imported 'mongoose' two times?

File: src/api/controller/search/searchController.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      console.log(error);
+      return res.status(500).send(messages.INTERNAL_SERVER_ERROR);
+    }
+  },
+  // method to get the data from the search collection.
+  searchData: async (req, res) => {
+    try {
+      let search = req.body.search;
+      const pageNumber = req.body.pageNumber;
+      const limit = req.body.limit || 5;
```
Comment (athanbhanit) [inline src/api/controller/search/searchController.js:24]: page limit is always be 10 by default.

File: src/api/middleware/joiSchema.js [hunk 3]
```diff
@@ -7,5 +7,10 @@ module.exports = {
         name : Joi.string().required(),
         author : Joi.string().required(),
         language : Joi.string().required()
+    }),
+    search : Joi.object({
+        search : Joi.string().required(),
+        pageNumber : Joi.number().min(1).required(),
+        limit : Joi.number()
```
Comment (athanbhanit) [inline src/api/middleware/joiSchema.js:14]: mark limit as optional

File: src/api/controller/create/createController.js (hunk 1 above)
Comment (rawalparas) [reply src/api/controller/create/createController.js:5]: I correct this.

File: src/api/controller/search/searchController.js (hunk 2 above)
Comment (rawalparas) [reply src/api/controller/search/searchController.js:24]: It is done.

File: src/api/middleware/joiSchema.js (hunk 3 above)
Comment (rawalparas) [reply src/api/middleware/joiSchema.js:14]: it is done

**Review by athanbhanit — APPROVED** [review]

**Pull request merged**